log/logfmt_logger.go

```diff
@@ -18,16 +38,45 @@ func NewLogfmtLogger(w io.Writer) Logger {
 }
 
 func (l logfmtLogger) Log(keyvals ...interface{}) error {
-	// The Logger interface requires implementations to be safe for concurrent
-	// use by multiple goroutines. For this implementation that means making
-	// only one call to l.w.Write() for each call to Log. We first collect all
-	// of the bytes into b, and then call l.w.Write(b).
-	b, err := logfmt.MarshalKeyvals(keyvals...)
-	if err != nil {
+	// Code copied from logfmt.MarshalKeyvals and modified
+	// to reuse encoders and buffers with sync.Pool and
+	// call EndRecord
```

Comment:
@ChrisHines Do you think it would make sense to have a `MarshalKeyvalsTo` (name for debate) in the logfmt package that takes an *Encoder instead of allocating a new one? This way `MarshalKeyvals` could be implemented by calling `MarshalKeyvalsTo` with a bytes.Buffer and we could avoid having to duplicate the code here.

Something like this 

```go
func MarshalKeyvalsTo(enc *Encoder, keyvals ...interface{}) error
```